src/pharmpy/tools/modelsearch/tool.py

```diff
@@ -59,21 +62,60 @@ def create_workflow(
     >>> run_modelsearch('ABSORPTION(ZO);PERIPHERALS(1)', 'exhaustive', results=results, model=model) # doctest: +SKIP
 
     """
-
     wb = WorkflowBuilder(name='modelsearch')
+    start_task = Task(
+        'start_modelsearch',
+        start,
+        search_space,
+        algorithm,
+        iiv_strategy,
+        rank_type,
+        cutoff,
+        results,
+        model,
+    )
+    wb.add_task(start_task)
+    task_results = Task('results', _results)
+    wb.add_task(task_results, predecessors=[start_task])
+    return Workflow(wb)
 
-    if model:
-        start_task = Task('start_modelsearch', start, model)
-    else:
-        start_task = Task('start_modelsearch', start)
 
+def start(
+    context,
+    search_space,
+    algorithm,
+    iiv_strategy,
+    rank_type,
+    cutoff,
+    results,
+    model,
+):
+    wb = WorkflowBuilder()
+
+    start_task = Task('start_modelsearch', _start, model)
     wb.add_task(start_task)
 
     algorithm_func = getattr(algorithms, algorithm)
 
-    mfl_statements = mfl_parse(search_space)
-    wf_search, candidate_model_tasks = algorithm_func(mfl_statements, iiv_strategy)
-    wb.insert_workflow(wf_search, predecessors=wb.output_tasks)
+    if isinstance(search_space, str):
+        mfl_statements = mfl_parse(search_space, mfl_class=True)
+    else:
+        mfl_statements = search_space
+
+    # Add base model task
+    base_task = Task("create_base_model", create_base_model, mfl_statements)
```

Comment:
Shouldn't this task only be done if the input model be done if the input and the base are different? To me it feels a bit counter-intuitive to create it and reroute the results from the input

Reply:
This was mainly done for consistency in the workflow but yes, I can change the workflow to only create a base model if needed